src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php

```diff
@@ -200,7 +201,12 @@ private function fetchMessage(): bool
 
         foreach ($this->currentResponse->getMessages() as $message) {
             $headers = [];
-            foreach ($message->getMessageAttributes() as $name => $attribute) {
+            $attributes = $message->getMessageAttributes();
+            if (isset($attributes[self::MESSAGE_ATTRIBUTE_NAME]) && 'String' === $attributes[self::MESSAGE_ATTRIBUTE_NAME]->getDataType()) {
+                $headers = \json_decode($attributes[self::MESSAGE_ATTRIBUTE_NAME]->getStringValue(), true);
+                unset($attributes[self::MESSAGE_ATTRIBUTE_NAME]);
+            }
+            foreach ($attributes as $name => $attribute) {
```

Comment:
This is need for BC compatibility for people that previously successfully encoded message without special char